application/src/main/java/org/thingsboard/server/service/security/auth/TokenOutdatingService.java

```diff
@@ -68,14 +62,36 @@ public boolean isOutdated(JwtToken token, UserId userId) {
                          * as all the tokens issued before the outdatage time
                          * are now expired by themselves
                          * */
-                        usersUpdateTimeCache.evict(toKey(userId));
+                        handleEvictEvent(new UsersUpdateTimeCacheEvictEvent(userId, sessionId));
                         return false;
                     }
                 })
                 .orElse(false);
     }
 
-    private String toKey(UserId userId) {
-        return userId.getId().toString();
+    @TransactionalEventListener(classes = UsersUpdateTimeCacheEvictEvent.class)
+    @Override
+    public void handleEvictEvent(UsersUpdateTimeCacheEvictEvent event) {
+        HashMap<String, Long> userSessions = cache.get(event.getUserId()).get();
```

Comment:
Do not store map with sessions as a value, use userId and sessionId as a key for the same cache and value as invalidation time